add su support introduced in logrotate 3.8.0

CharlieR-o-o-t opened the pull request.

alvagante commented on manifests/rule.pp:
```diff
@@ -337,6 +349,14 @@
 
+  case $su {
+    'undef',false: {}
+    true: { $_su = 'su' }
```
Variables names starting with _ are depracated and probably fail on Puppet 4

alvagante commented on templates/rule.erb:
```diff
@@ -18,6 +18,17 @@
     end
   end
 
+  if scope.to_hash.has_key?('_su')
```
This part looks a bit overkill, why not just doing something like:
<% if @su %>
su <%= @su_owner %>  <%= @su_group %> 
<% end>
without the need to set a @_su variable.
(am I missing something?)

CharlieR-o-o-t commented:
yes, I agree, it's overhead, I just merged realization from logrotate/rodjek. 

I have fixed everything what was in discussion and updated commit. 
Thanks

alvagante commented:
thx

CharlieR-o-o-t commented:
Sorry, I had mistype at rule.erb :(((
